### PR TITLE
fix 12441 [Accessibility] HC themes: The color contrast is less than 4.5:1 for the text "Document does not contain any pages" in printPreviewControl

### LINKED ISSUE #12441: [Accessibility] HC themes: The color contrast is less than 4.5:1 for the text "Document does not contain any pages" in printPreviewControl
### .NET version

.NET SDK 10.0.100-alpha.1.24556.12

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No, also repro in previous versions

### Issue description

The color contrast is less than 4.5:1 for the text "Document does not contain any pages" in printPreviewControl 

**Win11 Results:**
Image

**Win10 results:**
Image

**Framework results:**
Image

### Steps to reproduce

1. Use attached application:
[RepoApplication.zip](https://github.com/user-attachments/files/17655376/RepoApplication.zip)
2. Set windows to different HC themes
3. build and run application
4. Obserrve the text "Document does not contain any pages" in printPreviewControl by using Accessibility Insight tools.

### PR DESCRIPTION
**Pull request by Epica3055**
Fixes #12441

## Proposed changes

- 
- change back color and fore color
- 

### Before

Image

### After
image
image
image
image
image

## Test methodology 

- 
- manually accessibility insights
- 

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12448)

### CODE REVIEW

**Review by LeafShi1 — COMMENTED** [review]
LGTM！The test case `ShowPrintPreviewControl_BackColorIsCorrect` need to be updated.

File: src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintPreviewControl.cs [hunk 1]
```diff
@@ -592,7 +592,13 @@ private void CalculatePageInfo()
 
     private void DrawMessage(Graphics g, Rectangle rect, bool isExceptionPrinting)
     {
-        using var brush = SystemColors.ControlText.GetCachedSolidBrushScope();
+        Color brushColor = SystemColors.ControlText;
+        if (SystemInformation.HighContrast && Parent is not null)
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintPreviewControl.cs:596]: User pattern matching in order to access the Parent property only once - 
```suggestion
        if (SystemInformation.HighContrast && Parent is Control parent)
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Printing/PrintPreviewControlTests.cs [hunk 2]
```diff
@@ -8,34 +8,37 @@ namespace System.Windows.Forms.Tests;
...
+        Assert.Equal(SystemColors.ControlDarkDark.ToArgb(), actualBackColorArgb);
+        // Default AppWorkSpace color in HC theme does not allow to follow HC standards.
+        Assert.False(SystemColors.AppWorkspace.ToArgb().Equals(actualBackColorArgb));
+
+        control.BackColor = Color.Green;
+
+        actualBackColorArgb = control.TestAccessor().Dynamic.GetBackColor(true).ToArgb();
+
+        Assert.Equal(Color.Green.ToArgb(), actualBackColorArgb);
         // Default AppWorkSpace color in HC theme does not allow to follow HC standards.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/Printing/PrintPreviewControlTests.cs:41]: Consider removing this copy of this comment.

**Comment by Olina-Zhang** [comment]
Tested this PR fixing all HC modes on Windows 10 and 11, issue was fixed with no new issue found.

**Comment by Tanya-Solyanik** [comment]
Please backport it to release/9.0

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**